### ISSUE TITLE
[@types/react-filepond] shouldn't server urls be allowed to be explicitly null?

### ISSUE THREAD

**Issue by lamontadams** [DefinitelyTyped/DefinitelyTyped#35267]
- [x ] I tried using the `@types/react-filepond` package and had problems.
  - Authors: @zposten 

Hey @zposten - 

It looks like server urls are allowed to be null per the [Server documentation](https://pqina.nl/filepond/docs/patterns/api/server/#configuration) so shouldn't `FilePondServerConfigProps` actually be defined as something like...

```
interface FilePondServerConfigProps {
    instantUpload?: boolean;
    server?: string | {
        process: string | ServerUrl | null;
        revert: string | ServerUrl | null;
        restore: string | ServerUrl | null;
        load: string | ServerUrl | null;
        fetch: string | ServerUrl | null;
    };
}
```

I have an upload process that only needs the process method and the way the typedef is written now I can't null any of the other properties on `server`. Is there another way to go about this? Happy to do a PR, just want to make sure I'm not crazy before I try.

**Comment by zposten** [comment]
Hey @lamontadams,

No, I think what you're saying totally makes sense.  But to make it consistent with how everything else in the definition is done, would making the server properties optional work?

```ts
interface FilePondServerConfigProps {
    instantUpload?: boolean;
    server?: string | {
        process?: string | ServerUrl;
        revert?: string | ServerUrl;
        restore?: string | ServerUrl;
        load?: string | ServerUrl;
        fetch?: string | ServerUrl;
    };
}
```

If so, I can make a PR for this tomorrow, as there was one other change to this type definition that someone had asked about awhile ago that is still sitting in my inbox.

Thanks for catching this!

**Comment by zposten** [comment]
Okay great!  And I'm glad it has helped you 😄

**Comment by zposten** [comment]
This should be fixed in #35291

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
